Swift/Tests/ReplicatorTest+Collection.swift

```diff
@@ -0,0 +1,421 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+import CouchbaseLiteSwift
+
+class ReplicatorTest_Collection: ReplicatorTest {
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+    }
+
+    override func tearDownWithError() throws {
+        try super.tearDownWithError()
+    }
+    
+    // MARK: 8.13 Replicator Configuration
+    
+    func testCreateConfigWithDatabase() throws {
+        let url = URL(string: "wss://foo")!
+        let target = URLEndpoint(url: url)
+        
+        let config = ReplicatorConfiguration(database: self.db, target: target)
+        XCTAssertEqual(config.collections.count, 1)
+        
+        
+        guard let col = try self.db.defaultCollection() else {
+            XCTFail("Default collection is missing!")
+            return
+        }
+        
+        XCTAssertEqual(col, config.collections[0])
+        XCTAssertEqual(col.name, config.collections[0].name)
+        XCTAssertEqual(col.scope.name, config.collections[0].scope.name)
+        
+        guard let colConfig = config.collectionConfig(col) else {
+            XCTFail("Missing default collection")
+            return
+        }
+        
+        XCTAssertNil(colConfig.documentIDs)
+        XCTAssertNil(colConfig.channels)
+        XCTAssertNil(colConfig.conflictResolver)
+        XCTAssertNil(colConfig.pushFilter)
+        XCTAssertNil(colConfig.pullFilter)
+        
+        XCTAssertEqual(config.database.path, self.db.path)
+    }
+    
+    func collectionConfig(_ config: ReplicatorConfiguration, col: Collection) -> CollectionConfiguration {
+        guard let colConfig = config.collectionConfig(col) else {
+            fatalError("Collection config is missing!")
+        }
+        
+        return colConfig
+    }
+    
+    func defaultCollection() throws -> Collection {
+        guard let col = try self.db.defaultCollection() else {
+            fatalError("Default collection is missing!")
+        }
+        
+        return col
+    }
+    
+    func testConfigWithDatabaseAndConflictResolver() throws {
+        let url = URL(string: "wss://foo")!
+        let target = URLEndpoint(url: url)
+        
+        var config = ReplicatorConfiguration(database: self.db, target: target)
+        
+        let conflictResolver = TestConflictResolver({ (con: Conflict) -> Document? in
+            return con.remoteDocument
+        })
+        config.conflictResolver = conflictResolver
+        XCTAssertNotNil(conflictResolver.uid)
+        
+        
+        var colConfig = collectionConfig(config, col: try defaultCollection())
+        XCTAssertEqual(conflictResolver.uid, (config.conflictResolver as! TestConflictResolver).uid)
```

Comment:
As TestConflictResolver is a class, can you just compare b/w conflictResolver and config.conflictResolver without using uid? Basically, they will be the same object.

Reply:
Made the TestConflictResolver equatable, so now it can be checked for equality in swift. But its still same as checking `resolver.uid == resolver2.uid`.